### ISSUE TITLE
RPC communication failed with message: Cannot invoke RPC on closed channel!

### ISSUE THREAD

**Issue by uduse** [facebookresearch/CompilerGym#585]
## ❓ Questions and Help

Some part of my code has this:
```python
assert env.in_episode
env2 = env.fork()
```
and even though the assertion passes, the `env.fork()` fails, reporting `compiler_gym.service.connection.ServiceIsClosed: RPC communication failed with message: Cannot invoke RPC on closed channel!`

I'm asking because I assume that if `env.in_episode == True` then the `env` must be available for a `env.fork()`, is that right?

**Comment by ChrisCummins** [comment]
> I'm asking because I assume that if `env.in_episode == True` then the `env` must be available for a `env.fork()`, is that right?

Yes, that's right. The error message "Cannot invoke RPC on closed channel" could be cause by a few things but the most likely explanation is the backend service crashing during the call to `fork()`.

Could you please post a complete minimal code sample so that I can repro? It may would also be helpful if you could include the verbose logging output when run on your machine. To enable this, set environment variable `COMPILER_GYM_DEBUG=4`, and insert the following snippet at the start of your python script (before you create the environment):

```py
import logging
logging.basicConfig(level=logging.DEBUG)
```
